src/main/java/web/Servlet.java

```diff
@@ -0,0 +1,34 @@
+package web;
+
+import exception.OrderNotFoundException;
+import exception.UserNotFoundException;
+import web.command_pattern.CommandDistributor;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@WebServlet("/*")
+public class Servlet extends HttpServlet {
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+        try {
+            CommandDistributor.getInstance().getCommand(req.getPathInfo(), req.getMethod()).execute(req, resp);
+        } catch (UserNotFoundException e) {
+            resp.sendError(HttpServletResponse.SC_UNAUTHORIZED);
+        } catch (OrderNotFoundException e) {
+            resp.sendError(HttpServletResponse.SC_BAD_REQUEST);
+        }
+    }
+
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+        try {
+            CommandDistributor.getInstance().getCommand(req.getPathInfo(), req.getMethod()).execute(req, resp);
+        } catch (UserNotFoundException e) {
+            resp.sendError(HttpServletResponse.SC_UNAUTHORIZED);
+        } catch (OrderNotFoundException e) {
+            resp.sendError(HttpServletResponse.SC_BAD_REQUEST);
+        }
+    }
```

Comment:
1. Method bodies of the `doPost` and `doGet` methods look pretty similar...
2. Why do you return 401 status code if a user was not found, and 400 when an order was not found? Both these exceptions look more like `404 NOT FOUND`, but the actual status code depends on the service implementation.

Reply:
1. yet don't know how to change that, at last meeting it seems Nikita allowed to leave a duplicate

Reply:
1. As far as I remember, your CommandManager (don't remember the actual name) takes the http method into account during command resolution. Therefore, you can create a new method identical to your `doGet` or `doPost` method and call this new method from within the `doGet` and `doPost`.